.circleci/workflows.yml

```diff
@@ -119,9 +119,25 @@ jobs:
               - << pipeline.parameters.validate-bqetl >>
               - << pipeline.parameters.deploy >>
           steps:
+            - &skip_forked_pr
+              run:
+                name: Early return if this build is from a forked PR
+                command: |
+                  if [ -n "$CIRCLE_PR_NUMBER" ]; then
+                    echo "Cannot pass creds to forked PRs," \
+                      "so marking this step successful"
+                    circleci-agent step halt
+                  fi
             - checkout
             - *restore_venv_cache
             - *build
+            - &authenticate
+              run:
+                name: Authenticate to GCP
+                command: |
+                  export GOOGLE_APPLICATION_CREDENTIALS="/tmp/gcp.json"
+                  echo 'export GOOGLE_APPLICATION_CREDENTIALS="/tmp/gcp.json"' >> "$BASH_ENV"
+                  echo "$GCLOUD_SERVICE_KEY" > "$GOOGLE_APPLICATION_CREDENTIALS"
```

Comment:
The `test-sql`, `integration`, and `private-generate-sql` jobs don't set up authentication. Should they?

Reply:
`private-generate-sql` does indeed need this. At least the current tests in `integration` and `test-sql` don't depend on dryruns. So they don't need to change

Reply:
> At least the current tests in `integration` and `test-sql` don't depend on dryruns. So they don't need to change

But the `test-sql` job has forked-PR-detection logic that says "_Cannot pass creds to forked PRs_". Is that referring to different credentials?

Reply:
Yeah, those are different credentials afaik. for running some of these SQL tests we write test data to a BigQuery test project that needs these credentials